src/helpers/client/FormValue.js

```diff
@@ -0,0 +1,89 @@
+/** Handles the core of form values, including:
+ * - Storing both a current and future value
+ * - Tracking when the user manipulates a future value
+ * - Propagating current values to future values if they
+ *     haven't been manipulated by the user
+ * - Making this instance serializable
+ * - Making sure only valid values are saved.
+ * 
+ * All values stored in a FormValue instance should be valid
+ * 
+ * There must be a better way to write that.
+ * @todo Should `FormValue`s contain a reference to their
+ *     client object?
```

Comment:
If this were a database model, the answer would be yes. I think that's effectively what we're simulating, so probably.